pallets/automation-time/src/lib.rs

```diff
@@ -562,14 +562,14 @@ pub mod pallet {
 		/// * Get the most recent timestamp from the block.
 		/// * Convert the ms unix timestamp to seconds.
 		/// * Bring the timestamp down to the last whole hour.
-		pub fn get_current_time_slot() -> Result<UnixTime, Error<T>> {
+		pub fn get_current_time_slot() -> Result<UnixTime, DispatchError> {
 			let now = <timestamp::Pallet<T>>::get().saturated_into::<UnixTime>();
 			if now == 0 {
 				Err(Error::<T>::BlockTimeNotSet)?
 			}
-			let now = now / 1000;
-			let diff_to_hour = now % 3600;
-			Ok(now.saturating_sub(diff_to_hour))
+			let now = now.checked_div(1000).ok_or(ArithmeticError::Overflow)?;
+			let diff_to_hour = now.checked_rem(3600).ok_or(ArithmeticError::Overflow)?;
```

Comment:
doesn't checked_* functions return Option<number>? do you have to unwrap that option?

Reply:
Doesn't checked_* functions return Option? 
checked_div return Option: `Some()` or `None`.

do you have to unwrap that option?
Yes, I have to unwrap the option.

When it is None, dispatch ArithmeticError to tell the user that an error was encountered during processing.
Here is the code in substrate, which I referenced.

https://github.com/paritytech/substrate/blob/6a946fc36d68b89599d7ca1ab03803d10c78468c/frame/assets/src/functions.rs#L71

https://github.com/paritytech/substrate/blob/6a946fc36d68b89599d7ca1ab03803d10c78468c/frame/balances/src/lib.rs#L1033

Reply:
ahh you're good! my bad, i missed the question mark